pkg/operator/admission/webhook.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/ceph/csi"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"os"
+)
+
+const (
+	appName            = "rook-ceph-admission-controller"
+	secretVolumeName   = "webhook-certificates"
+	serviceAccountName = "rook-ceph-admission-controller"
+	portName           = "webhook-api"
+)
+
+var (
+	validatePath       = "/validate"
+	servicePort  int32 = 443
+	serverPort   int32 = 8079
+	namespace          = os.Getenv(k8sutil.PodNamespaceEnvVar)
+)
+
+func isSecretPresent(context *clusterd.Context) (bool, error) {
+	logger.Infof("looking for secret %q", appName)
+	_, err := context.Clientset.CoreV1().Secrets(namespace).Get(appName, metav1.GetOptions{})
+	if err != nil {
+		// If secret is not found. All good ! Proceed with rook without admission controllers
+		if apierrors.IsNotFound(err) {
+			logger.Infof("secret %q not found. proceeding without the admission controllers", appName)
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func createWebhookService(context *clusterd.Context) error {
+	webhookService := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      appName,
+			Namespace: namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Port: servicePort,
+					TargetPort: intstr.IntOrString{
+						IntVal: serverPort,
+					},
+				},
+			},
+
+			Selector: map[string]string{
+				k8sutil.AppAttr: appName,
+			},
+		},
+	}
+
+	_, err := k8sutil.CreateOrUpdateService(context.Clientset, namespace, &webhookService)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+	return nil
+}
+
+// InitWebhookIfSecretPresent will initialize the webhook if secret is detected
+func InitWebhookIfSecretPresent(context *clusterd.Context, admissionImage string) error {
```

Comment:
Calling it the admission controller is more intuitive to me than calling it the webhook. What do you think?
```suggestion
func StartControllerIfSecretPresent(context *clusterd.Context, admissionImage string) error {
```